e2e/playwright/actions/src/tests/library/library-actions.spec.ts

```diff
@@ -0,0 +1,444 @@
+import { Site } from '@alfresco/js-api';
+
+test.describe('Library actions ', () => {
+  const apiClientFactory = new ApiClientFactory();
+  const notMemberString = 'Not a member';
+  const managerRole = 'Manager';
+  const contributorRole = 'Contributor';
+  const consumerRole = 'Consumer';
+  const joinButton = 'Join';
+  const libraryJoinedMessage = 'Library joined';
+  const requestToJoinMessage = 'Request sent to join this library';
+  const cancelJoinRequestButton = 'Cancel Join Request';
+  const cancelJoinRequestMessage = 'Canceled the request to join the library';
+  const leaveLibraryButton = 'Leave Library';
+  const leaveDialogTitle = 'Leave this library?';
+  const leaveDialogContent = 'Leaving will remove your access.';
+  const leftMessage = 'You have left the library';
+  const favoriteButton = 'Favorite';
+  const removeFavoriteButton = 'Remove Favorite';
+  const deleteButton = 'Delete';
+  const libraryDeletedMessage = 'Library deleted';
+  const siteString = 'site';
+  const loadString = 'load';
+  const domContentLoadedString = 'domcontentloaded';
+
+  const username1 = `user-${Utils.random()}`;
+  const username2 = `user-${Utils.random()}`;
+  const adminLibrary1 = `playwright-A-library-${Utils.random()}`;
+  const adminLibrary2 = `playwright-A-library-${Utils.random()}`;
+  const adminLibrary3 = `playwright-A-library-${Utils.random()}`;
+  const adminLibrary4 = `playwright-A-library-${Utils.random()}`;
+  const adminModerateLibrary1 = `playwright-Am-library-${Utils.random()}`;
+  const adminModerateLibrary2 = `playwright-Am-library-${Utils.random()}`;
+  const adminModerateLibrary3 = `playwright-Am-library-${Utils.random()}`;
+  const adminModerateLibrary4 = `playwright-Am-library-${Utils.random()}`;
+  const user1Library1 = `playwright-U1-library-${Utils.random()}`;
+  const user1Library2 = `playwright-U1-library-${Utils.random()}`;
+  const user1Library3 = `playwright-U1-library-${Utils.random()}`;
+  const user1Library4 = `playwright-U1-library-${Utils.random()}`;
+  const user1Library5 = `playwright-U1-library-${Utils.random()}`;
+  const user2Library1 = `playwright-U2-library-${Utils.random()}`;
+  const user2Library2 = `playwright-U2-library-${Utils.random()}`;
+  const user2Library3 = `playwright-U2-library-${Utils.random()}`;
+  const user2Library4 = `playwright-U2-library-${Utils.random()}`;
+  const user2Library5Delete = `playwright-T-library-${Utils.random()}`;
+  const user2Library6Delete = `playwright-T-library-${Utils.random()}`;
+  const user2Library7Delete = `playwright-T-library-${Utils.random()}`;
+
+  const adminLibraryIds: string[] = [adminLibrary1, adminLibrary2, adminLibrary3, adminLibrary4];
+  const adminModerateLibraryIds: string[] = [adminModerateLibrary1, adminModerateLibrary2, adminModerateLibrary3, adminModerateLibrary4];
+  const user1LibraryIds: string[] = [user1Library1, user1Library2, user1Library3, user1Library4, user1Library5];
+  const siteRoles: string[] = [
```

Comment:
no need for type

Reply:
done in [aa865dc](https://github.com/Alfresco/alfresco-content-app/pull/3545/commits/aa865dcf243348da373c4560117ab99260fb3fe8)